### PR TITLE
DataHub service-level agreement

### PR DESCRIPTION
**Pull request by eickler**

### CODE REVIEW

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+{{< product-c8y-iot >}} DataHub is a component of the {{< product-c8y-iot >}} platform that enables efficient long-term storage and analysis of IoT data. It offloads data from the operational store to a data lake, allowing for scalable SQL-based querying via standard interfaces like ODBC and JDBC.
+
+This agreement defines the service level of {{< product-c8y-iot >}} Software-as-a-Service operated by {{< company-c8y >}}. {{< product-c8y-iot >}} Software-Operated-as-a-Service and {{< product-c8y-iot >}} Edge are outside the scope of this agreement.
+
+### Service features
+
+{{< product-c8y-iot >}} DataHub provides the following features.
+
+* **Scalable and economic long-term data storage:** {{< product-c8y-iot >}} DataHub offloads data into economic data lake storage for long-term data retention, permitting you to adapt the retention times of your operational store.
```
Comment (cheinz-sag) [inline content/service-terms/service-level-bundle/datahub-sla.md]: long-term data retention -> long-term archival

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+* **Advanced data querying:** Data is made available for in-depth analysis to SQL-based analytics tools such as business intelligence, notebook and dashboarding applications.
+* **Configurable offloading:** So-called "offloaders" permit you to select what data is offloaded and how it is mapped into the data lake for user-friendly querying.
```
Comment (cheinz-sag) [inline content/service-terms/service-level-bundle/datahub-sla.md]: **Easy-to-use data pipelines**: So-called offloading configuration permit you ... user-friendly, SQL-based querying.

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+{{< product-c8y-iot >}} DataHub is a component of the {{< product-c8y-iot >}} platform that enables efficient long-term storage and analysis of IoT data. It offloads data from the operational store to a data lake, allowing for scalable SQL-based querying via standard interfaces like ODBC and JDBC.
+
+This agreement defines the service level of {{< product-c8y-iot >}} Software-as-a-Service operated by {{< company-c8y >}}. {{< product-c8y-iot >}} Software-Operated-as-a-Service and {{< product-c8y-iot >}} Edge are outside the scope of this agreement.
+
+### Service features
+
+{{< product-c8y-iot >}} DataHub provides the following features.
+
+* **Scalable and economic long-term data storage:** {{< product-c8y-iot >}} DataHub offloads data into economic data lake storage for long-term data retention, permitting you to adapt the retention times of your operational store.
+* **Advanced data querying:** Data is made available for in-depth analysis to SQL-based analytics tools such as business intelligence, notebook and dashboarding applications.
```
Comment (cheinz-sag) [inline content/service-terms/service-level-bundle/datahub-sla.md]: Long-term data is made available ...

File: content/service-terms/service-level-bundle/datahub-sla.md (hunk 1 above)
Comment (cheinz-sag) [inline content/service-terms/service-level-bundle/datahub-sla.md]: ... into economic and scalable ...

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+* **Configurable offloading:** So-called "offloaders" permit you to select what data is offloaded and how it is mapped into the data lake for user-friendly querying.
+* TBD: What else to mention here as main points?
```
Comment (cheinz-sag) [inline content/service-terms/service-level-bundle/datahub-sla.md]: Guess that's it

File: content/service-terms/service-level-bundle/datahub-sla.md (hunk 1 above)
Comment (eickler) [reply content/service-terms/service-level-bundle/datahub-sla.md]: Not sure which one is actually better ... "retention" might imply that you can still use it from the C8Y user interface (which I heard as longer-term idea) and "archival" implies that you can restore it (which is not a use cases AFAIK). I'll think a bit more about the terms.

Comment (eickler) [reply content/service-terms/service-level-bundle/datahub-sla.md]: Did a workaround :-)

File: content/service-terms/service-level-bundle/datahub-sla.md (hunk 2 above)
Comment (eickler) [reply content/service-terms/service-level-bundle/datahub-sla.md]: Used the term from the customer docs.

File: content/service-terms/service-level-bundle/datahub-sla.md (hunk 1 above)
Comment (eickler) [reply content/service-terms/service-level-bundle/datahub-sla.md]: Yeah well, that would imply that the operational store is not scalable, so I'd rather not add that here.

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+* **Advanced data querying:** Long-term data is made available for in-depth analysis to SQL-based analytics tools such as business intelligence, notebook, and dashboarding applications.
+* **Configurable offloading:** So-called "offloading pipelines" permit you to select what data is offloaded and how it is mapped into the data lake for user-friendly, SQL-based querying.
+
+### Customer responsibilities
+
+Customers are encouraged to review the [{{< product-c8y-iot >}} DataHub documentation](/datahub). In particular, Customer acknowledges the following Customer responsibilities:
+
+* **Data lake provisioning:** Customer provides data lake storage. Customer is responsible for setting correct storage permissions and configuring correct credentials for data lake access in {{< product-c8y-iot >}} DataHub. Please refer to Section ["Permissions for data lake and space"](/datahub/setting-up-datahub/#permissions-for-data-lake-and-space) for more information. We recommend provisioning the data lake in the same hyperscale and hyperscaler region as Customer's {{< product-c8y-iot >}} tenants for the best performance.
```
Comment (BeateRixen) [inline content/service-terms/service-level-bundle/datahub-sla.md]:
```suggestion
* **Data lake provisioning:** Customer provides data lake storage. Customer is responsible for setting correct storage permissions and configuring correct credentials for data lake access in {{< product-c8y-iot >}} DataHub. Refer to [Permissions for data lake and space](/datahub/setting-up-datahub/#permissions-for-data-lake-and-space) for more information. We recommend provisioning the data lake in the same hyperscale and hyperscaler region as Customer's {{< product-c8y-iot >}} tenants for the best performance.
```

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+* **Storage cost:** Customer is responsible for managing data retention policies within Customer's configured S3 bucket or Azure Data Lake Storage and ensuring that offloading jobs are configured appropriately to align with Customer's organizational requirements, data management strategies and budgets.
+* **Offloading configuration:** Customer maintains compatibility of offloading configurations with the actual data structures present in the operational store. For more information, please see Section ["Aligning data modeling and offloading"](/datahub/working-with-datahub/#aligning-data-modeling-and-offloading) and Section ["Dealing with mixed types"](/datahub/working-with-datahub/#mixed-types).
```
Comment (BeateRixen) [inline content/service-terms/service-level-bundle/datahub-sla.md:31]:
```suggestion
* **Offloading configuration:** Customer maintains compatibility of offloading configurations with the actual data structures present in the operational store. For more information, see [Aligning data modeling and offloading](/datahub/working-with-datahub/#aligning-data-modeling-and-offloading) and [Dealing with mixed types](/datahub/working-with-datahub/#mixed-types).
```

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+* **Offloading monitoring:** To avoid data loss, Customer is recommended to monitor and respond to offloading alarms as data loss may occur. For example, if the data structures in the operational store change, offloaders configured to ["stop pipeline"](/datahub/working-with-datahub/#mixed-types) will halt. If the offloaders are not reconfigured before the configured retention intervals in the operational store apply, data may be deleted before it is offloaded. For this reason, Customer is advised to configure suitable retention intervals in the operational store to allow for reaction times. Alarms can be, for example, forwarded to email for better visibility.
+* **Data lake modifications:** Customer is responsible for Customer's modifications to the data lake, such as moving or deleting files, as outlined in Section ["Modifying data in the data lake"](/datahub/working-with-datahub/#modifying-data-in-the-data-lake). In particular, moving files may break DataHub offloading jobs.
```
Comment (BeateRixen) [inline content/service-terms/service-level-bundle/datahub-sla.md:33]:
```suggestion
* **Data lake modifications:** Customer is responsible for Customer's modifications to the data lake, such as moving or deleting files, as outlined in [Modifying data in the data lake](/datahub/working-with-datahub/#modifying-data-in-the-data-lake). In particular, moving files may break DataHub offloading jobs.
```

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+* **Production environments:** 99.00% availability
+* **Preproduction environments:** 95.00% availability
+
+Service availability for [{{< product-c8y-iot >}} DataHub APIs](https://www.cumulocity.com/api/datahub/#tag/Standard-API) is calculated as outlined in the [platform's service availability section](/service-terms/service-level/#service-availability).
+
+Offloading jobs may not run at a scheduled time if a previous offloading job is still in progress (for example, due to an initial larger volume upload or after a longer period of inactivity) or during scheduled maintenance. Subsequent offloading jobs will eventually catch up with remaining data.
+
+### Support and maintenance
+
+Support and maintenance are provided as outlined in the [platform service-level agreement](/service-terms/service-level/#support-and-maintenance).
```
Comment (BeateRixen) [inline content/service-terms/service-level-bundle/datahub-sla.md:60]:
```suggestion
Support and maintenance are provided as outlined in the [Platform service-level agreement](/service-terms/service-level/#support-and-maintenance).
```

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+{{< product-c8y-iot >}} DataHub provides the following features.
+
+* **Scalable and economic long-term data storage:** {{< product-c8y-iot >}} DataHub offloads data into economic data lake storage outside of the operational store for long-term data retention, permitting you to shorten the retention times of the more costly operational store.
+* **Advanced data querying:** Long-term data is made available for in-depth analysis to SQL-based analytics tools such as business intelligence, notebook, and dashboarding applications.
+* **Configurable offloading:** So-called "offloading pipelines" permit you to select what data is offloaded and how it is mapped into the data lake for user-friendly, SQL-based querying.
+
+### Customer responsibilities
+
+Customers are encouraged to review the [{{< product-c8y-iot >}} DataHub documentation](/datahub). In particular, Customer acknowledges the following Customer responsibilities:
```
Comment (BeateRixen) [inline content/service-terms/service-level-bundle/datahub-sla.md:27]: For consistency, I would probably also use "Customer is encouraged" here

File: content/service-terms/quotas.md [hunk 10]
```diff
@@ -75,3 +76,15 @@ The quotas listed here reflect the maximum values for the cloud subscriptions un
...
+| ---------------------------------------------------------------------------------------------------------------------------------- | ---- | ---------: |
+| Number of offloaders per tenant                                                                                                    | Soft |        100 |
+| [Offloading frequency](/datahub/working-with-datahub/#configure-additional-settings)                                               | Hard |     hourly |
+| [Offloaded leaf properties](https://docs.dremio.com/current/get-started/cluster-deployments/architecture/limits/#catalog)          | Hard |       6400 |
+| Query time out                                                                                                                     | Hard |        ??? |
+| Query job retention                                                                                                                | Hard |      1 day |
+| [Rows in a query job](https://docs.dremio.com/current/get-started/cluster-deployments/architecture/limits/#execution)              | Hard |    1000000 |
```
Comment (neuerburg) [inline content/service-terms/quotas.md]: What does the limit on Offloaded leaf properties, Rows in a query job and ows in a high performance query mean to a customer?
Currently, they link to the Dremio documentation but do not explain what to consider when setting up offloading jobs within Cumulocity IoT.

We should describe somewhere in the Cumulocity documentation what these limits mean to Cumulocity users and use-cases and how a user can ensure to stay below the limits when e.g. creating a new offloading job.

File: content/service-terms/service-level-bundle/datahub-sla.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+  - /datahub-sla/
+---
+
+This agreement is made between {{< company-c8y >}} ("Provider") and the Customer ("Customer") who utilizes {{< product-c8y-iot >}} DataHub ("Service") for offloading and analyzing Internet of Things ("IoT") data using Provider's cloud instances ("software-as-a-service", "SaaS").
+
+### Service description
+
+{{< product-c8y-iot >}} DataHub is a component of the {{< product-c8y-iot >}} platform that enables efficient long-term storage and analysis of IoT data. It offloads data from the operational store to a data lake, allowing for scalable SQL-based querying via standard interfaces like ODBC and JDBC.
+
+This agreement defines the service level of {{< product-c8y-iot >}} Software-as-a-Service operated by {{< company-c8y >}}. {{< product-c8y-iot >}} Software-Operated-as-a-Service and {{< product-c8y-iot >}} Edge are outside the scope of this agreement.
```
Comment (neuerburg) [inline content/service-terms/service-level-bundle/datahub-sla.md]: Software Operated as a Service is not a standard industry term and should be explained or better not mentioned at all.

Also, I noticed that we have the same sentence stated in the Platform SLA but not in the Microservice Hosting SLA. For consistency, we should use the same wording in all three SLA descriptions.

File: content/service-terms/service-level-bundle/datahub-sla.md (hunk 5 above)
Comment (neuerburg) [reply content/service-terms/service-level-bundle/datahub-sla.md]: Small typo: "hyperscale and" => "hyperscaler and" or remove all together

File: content/service-terms/service-level-bundle/datahub-sla.md (hunk 9 above)
Comment (eickler) [reply content/service-terms/service-level-bundle/datahub-sla.md:27]: @BeateRixen The difference is: They should read the documentation, but they must acknowledge the responsibilities.

File: content/service-terms/quotas.md (hunk 10 above)
Comment (eickler) [reply content/service-terms/quotas.md]: Yeah, I agree. I would do that as a different PR to get this out first.

File: content/service-terms/service-level-bundle/datahub-sla.md (hunk 11 above)
Comment (eickler) [reply content/service-terms/service-level-bundle/datahub-sla.md]: I think I have also misunderstood the implication of SoaS. I thought it's managed services.

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 12]
```diff
@@ -0,0 +1,27 @@
+---
+date: 2024-12-05
+title: Further improvements to Service Terms
+change_type:
+  - value: change-2c7RdTdXo4
+    label: Improvement
+product_area: Platform services
+component:
+  - value: component-9vjGQz8Ag
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
  - value: component-docs-123
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    label: Core platform
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
    label: Documentation
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+date: 2024-12-05
+title: Further improvements to Service Terms
+change_type:
+  - value: change-2c7RdTdXo4
+    label: Improvement
+product_area: Platform services
+component:
+  - value: component-docs-123
+    label: Documentation
+  - value: component-A8vMaVaTg
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+date: 2024-12-05
+title: Further improvements to Service Terms
+change_type:
+  - value: change-2c7RdTdXo4
+    label: Improvement
+product_area: Platform services
+component:
+  - value: component-docs-123
+    label: Documentation
+    label: DataHub
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 16]
```diff
@@ -0,0 +1,25 @@
...
+  - value: change-2c7RdTdXo4
+    label: Improvement
+product_area: Platform services
+component:
+  - value: component-docs-123
+    label: Documentation
+build_artifact:
+  - value: tc-QHwMfWtBk7
+    label: cumulocity
+  - value: tc-H-tuq-8Es
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 17]
```diff
@@ -0,0 +1,24 @@
...
+  - value: change-2c7RdTdXo4
+    label: Improvement
+product_area: Platform services
+component:
+  - value: component-docs-123
+    label: Documentation
+build_artifact:
+  - value: tc-QHwMfWtBk7
+    label: cumulocity
+    label: cdh
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+change_type:
+  - value: change-2c7RdTdXo4
+    label: Improvement
+product_area: Platform services
+component:
+  - value: component-docs-123
+    label: Documentation
+build_artifact:
+  - value: tc-QHwMfWtBk7
+    label: cumulocity
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+title: Further improvements to Service Terms
+change_type:
+  - value: change-2c7RdTdXo4
+    label: Improvement
+product_area: Platform services
+component:
+  - value: component-docs-123
+    label: Documentation
+build_artifact:
+  - value: tc-QHwMfWtBk7
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
```

File: content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+component:
+  - value: component-docs-123
+    label: Documentation
+build_artifact:
+---
+The section on [Service terms](/service-terms/) has been further enhanced. It now also contains
+
+* the [Support service-level agreement](/service-terms/service-level/#support-sla),
+* the [DataHub service-level agreement](/service-terms/service-level/#datahub-sla),
+* the [Compatibility policy](/service-terms/compatibility-policy/) (moved from "Getting started"),
```
Comment (BeateRixen) [inline content/change-logs/platform-services/cumulocity-undefined-Service-Terms.md]:
```suggestion
* the [Compatibility policy](/service-terms/compatibility-policy/) (moved from "Technical concepts"),
```

**Review by BeateRixen — APPROVED** [review]

File: content/service-terms/quotas.md (hunk 10 above)
Comment (eickler) [reply content/service-terms/quotas.md]: I fixed now the links and added an explanation.

**Comment by eickler** [comment]
@neuerburg @BeateRixen I added the remaining review comments that came in via email. Can we merge the PR?

**Review by jvi-c8y — APPROVED** [review]
Looks good !

**Pull request merged**